### ISSUE TITLE
Create docker image

### ISSUE THREAD

**Issue by tkellogg** [tkellogg/fossil#5]
Create a docker image that runs `server.py` via uvicorn on all host names, exposes the port, etc. I'll push the image to docker central.

**Comment by aryannargotra** [comment]
Can you assign this to me

**Comment by tkellogg** [comment]
By the way, last night I changed it to use models via [llm](https://llm.datasette.io/en/stable/index.html). `llm` has a configuration file where it configures things like API keys and downloaded models. The directory that holds all that can be overridden via the `LLM_USER_PATH` environment variable (see [here](https://github.com/simonw/llm/blob/main/llm/__init__.py#L188-L195). I'm unsure how well this is documented, I just found that by looking in the code.

**Comment by aryannargotra** [comment]
Hi Tim,
I read your blog about Mastodon, and it's been so long since I truly liked something and found it really meaningful. It's such a great experience, and I'll try to contribute to this cause and learn new things if I have the opportunity.

I've changed jobs three times in the past year because I wasn't finding anything interesting in my work, which led to a loss of interest in the tech field. However, your project has really got me excited.

**Comment by aryannargotra** [comment]
![Screenshot 2024-01-05 163943](https://github.com/tkellogg/fossil/assets/44235198/f1a13670-930f-40a5-b66b-004c85002296)

Having issue installing the 3.10 version in my container

**Comment by tkellogg** [comment]
i wish it was that easy to install a new version of python, but oh, no, it truly sucks. Here's a couple easier methods

* [conda](https://saturncloud.io/blog/updating-python-to-a-specific-version-using-conda-a-guide/)
* [pyenv](https://github.com/pyenv/pyenv)

Honestly, just change `pyproject.toml` to allow python >=3.9

or rather, since you're using docker anyway, use the [python3.10](https://hub.docker.com/_/python) base image

**Comment by tkellogg** [comment]
Also, since you're in docker anyway, don't bother with the virtual environment. Docker provides all the isolation you need

**Comment by aryannargotra** [comment]
I am Using this DosckerFile  but getting out of memory error while building it 

![image](https://github.com/tkellogg/fossil/assets/44235198/92d41de4-37c6-4dfc-97ae-5bf73616bbd5)

**Comment by tkellogg** [comment]
Out of memory while building? 🤔  if you're out of docker cache space you can look into clearing it, that does happen a lot (although you seem to know what you're doing, so maybe you tried that already). If you're out of RAM, I'm not sure what to say, none of this is particularly big. I've used the `python` image before and it's quite small. `llm` does install a lot of packages, but I didn't think that really consumed much RAM, per se.

What stage does it die on?

**Comment by aryannargotra** [comment]
around here when it checks for vulnerabilities, it gets stuck.

![image](https://github.com/tkellogg/fossil/assets/44235198/d87a98a4-3784-48e6-a959-da25973f395d)

**Comment by tkellogg** [comment]
Are you talking about the "There is not enough space on the disk" error? You might need to increase the space available to the Docker cache. Regardless, that's a giant image, are we really using all that?

**Comment by aryannargotra** [comment]
Do we also need to config the openAI key?, Its in the config file. 
I've only configured the access token yet

**Comment by tkellogg** [comment]
i don't think you want any secrets in the image itself. i imagine you'd pull them in via `-e` environment variables or import the user's config directory via a `-v` volume at runtime.

i mentioned this before, but `llm` stores all it's configuration in your home directory, but you can override where it's stored. if you didn't import a file location as a volume, then changes made at runtime would be ephemeral (and you couldn't distribute the image because there's secrets in there).

btw, you should probably revoke the access token you posted a couple screenshots ago, just to be safe.

**Comment by tkellogg** [comment]
Docker is for immutable infrastructure, but SQLite and config mutate it. You might want to look up how people package up databases, like postgres
